wikimedia/wikipedia-ios
# Consider web banner preferences before displaying campaign modal

**Phabricator:** https://phabricator.wikimedia.org/T352306

### Notes
This PR fetches the web banner preferences upon login and checks them before displaying campaign modals. If there's any issue and we don't know their preferences, we display the campaign modal by default.

### Test Steps
1. On a device, set Japan as Region and set English as language. Change device date to May 30.
2. On desktop, log into your account on EN Wiki and uncheck Fundraising in banner preferences.
3. On device, fresh install on Staging. Background and foreground once on Explore feed, then pull to refresh to trigger campaign fetch.
4. Visit article on EN Wiki, confirm fundraising modal does NOT show.
5. Check preference on web. 
6. Repeat steps 3-4 and confirm you do see a modal.
